### PR TITLE
Add referenced FAQ to volunteer support notifs

### LINKED ISSUE #1715: E-mailed Support notifications do not include referenced FAQs
e.g. my notification e-mail for my visible answer at https://www.dreamwidth.org/support/see_request?id=32907 reads:

> A follow-up to the following  support request has been submitted:
> 
> Category: Entries
> Subject: Only 14 days worth of posts?
> User: kaberett
> URL:http://www.dreamwidth.org/support/see_request?id=32907
> Type: Screened Answer
> Text:
> 
> Hi stormcloude,
> 
> As explained in the linked FAQ and as you've observed, the Reading Page
> displays only the most recent 1000 entries from the last 14 days. These
> limits are the same as those on LiveJournal
> (http://www.livejournal.com/support/faq/219.html), for the same reasons of
> issues with server load and site performance. It is possible to bypass
> these limits by filtering the Reading Page by date
> (https://www.dreamwidth.org/support/faqbrowse?faqid=76), but this is a
> feature that is only available to paid accounts on both Dreamwidth and
> LiveJournal because of the demands it places on the servers.
> 
> As such there isn't any setting you can change, other than purchasing paid
> time. Details on how much paid accounts cost are available at
> https://www.dreamwidth.org/support/faqbrowse?faqid=153, and a summary of
> additional features available to paid account holders is provided at
> https://www.dreamwidth.org/support/faqbrowse?faqid=4. Alternatively, you
> can visit individual journals or communities to read their entire archives.
> 
> 
> I'm sorry that this wasn't what you were hoping to hear, and please do let
> me know if you have any further questions.
> 
> Best,
> -kaberett.
> 
> 
> ====
> 
> You can view this request here:
> 
> http://www.dreamwidth.org/support/see_request?id=32907
> 
> You are receiving this email because you've requested notifications of
> changes to support requests. You may change this notification setting here:
> 
> http://www.dreamwidth.org/support/changenotify

There's no indication that https://www.dreamwidth.org/support/faqbrowse?faqid=25 was included as a Reference FAQ, and it would be useful for record-keeping purposes not to have to click through to see this. I suspect this would also be useful from the perspective of users. I suggest adding a line to the effect of:

> FAQ Reference: What is my reading page? (https://www.dreamwidth.org/support/faqbrowse?faqid=25&view=full)

... for plaintext e-mails, with the FAQ title linkified for HTML e-mails.

### PR DESCRIPTION
**Pull request by kaberett**
Fixes #1715.

If a FAQ was referenced in a Support answer, the ticket update
notification e-mail will now include the FAQ title and link.

### CODE REVIEW

**Comment by kaberett** [comment]
NB I've split out the generation of the body of the e-mail for updates to put sorting-out-the-faqref in its own block of code, per sending out e-mail to the ticket submitter (see line 1000 onward). I'm not convinced this is particularly tidy, though, and the code repetition makes me wince; but the only way I can think of to do it while keeping generating-the-body-of-the-email-in-one-block is to include "FAQ REFERENCE: [none]" by default if no FAQ is referenced.

Thoughts/feedback/preferences v welcome!

File: cgi-bin/LJ/Support.pm [hunk 1]
```diff
@@ -1152,15 +1153,39 @@ sub work {
...
+            $lang = LJ::Support::prop( $spid, 'language' )
+                if LJ::is_enabled( 'support_request_language' );
+            $lang ||= $u->prop( 'browselang' ) if $u;
+            $lang ||= $LJ::DEFAULT_LANG;
+            
+            # now actually get the FAQ
+            my $faq = LJ::Faq->load( $faqid, lang => $lang );
+            if ( $faq ) {
+                $faq->render_in_place;
+                my $faqref = $faq->question_raw . " " . "$LJ::SITEROOT/support/faqbrowse?faqid=$faqid&view=full";
```
Comment (zorkian) [inline cgi-bin/LJ/Support.pm]: Aw does `LJ::Faq` not have a `url_full` method or something? Bummer.

File: cgi-bin/LJ/Support.pm [hunk 2]
```diff
@@ -1152,15 +1153,39 @@ sub work {
...
+                # now add it to the e-mail!
+                $body .= "\n" . LJ::Lang::ml( "support.email.notif.update.body.faqref", {
+                        faqref => $faqref
+                } );
+                $body .= "\n";
+            }
+        }
+        $body .= LJ::Lang::ml( "support.email.notif.update.body.text", {
+                text => $resp
+           } );
```
Comment (zorkian) [inline cgi-bin/LJ/Support.pm]: This should be dedented to match up with `$body`.

File: cgi-bin/LJ/Support.pm [hunk 3]
```diff
@@ -1152,15 +1153,39 @@ sub work {
...
-                type => $response_type,
                 username => LJ::trim( $show_name ),
                 url => "$LJ::SITEROOT/support/see_request?id=$spid",
-                text => $resp
+                type => $response_type
             } );
+        if ( $faqid ) {
+            # need to set up $lang
+            my $lang;
+            my $u = LJ::load_userid( $posterid ) if $posterid;
```
Comment (zorkian) [inline cgi-bin/LJ/Support.pm]: So, `my $x = foo if $bar;` is a trap in some versions of Perl, so we don't allow it. See: http://wiki.dreamwidth.net/wiki/index.php/Dev_Programming_Guidelines#Postfix_conditionals

You can just do `my ( $lang, $u );` above and then `$u = LJ::load_userid( $posterid ) if $posterid;` in this line.

File: cgi-bin/LJ/Support.pm (hunk 1 above)
Comment (kaberett) [inline cgi-bin/LJ/Support.pm]: This is how it's done in 1011 of LJ::Support too and I've just had a look and there super isn't a url_full method. Want me to file the bug? :-)

File: cgi-bin/LJ/Support.pm (hunk 2 above)
Comment (kaberett) [inline cgi-bin/LJ/Support.pm]: oops.

**Comment by kaberett** [comment]
aaaand done!

File: cgi-bin/LJ/Support.pm (hunk 3 above)
Comment (kaberett) [inline cgi-bin/LJ/Support.pm]: Thanks for the link!

**Comment by kaberett** [comment]
oh dammit, meant to rebase. keep forgetting that step. please hold ;)

File: cgi-bin/LJ/Support.pm (hunk 1 above)
Comment (zorkian) [inline cgi-bin/LJ/Support.pm]: It's a good babydev thing, so sure! It's a nice obvious kind of cleanup that would improve the readability of the code, IMO.

**Comment by zorkian** [comment]
Thanks for fixing it up! <3

**Pull request merged**